clients/client/src/main/java/org/projectnessie/client/StreamingUtil.java

```diff
@@ -63,25 +76,28 @@ private StreamingUtil() {
    * <p>The {@link Stream} returned by {@code getEntriesStream(ref, OptionalInt.empty())}, if not
    * limited, returns all commit-log entries.
    *
-   * @param ref a named reference (branch or tag name) or a commit-hash
+   * @param api The {@link NessieApiV1} to use
+   * @param ref a named reference (branch or tag name).
+   * @param builderCustomizer a Function that takes and returns {@link GetEntriesBuilder} that is
+   *     passed to the caller of this method. It allows the caller to customize the
+   *     GetEntriesBuilder with additional parameters (e.g.: hashOnRef, filter).
+   * @param maxRecords - a maximum number of records in the stream. If it is {@code
+   *     Optional.empty()} the stream will be unbounded.
    * @return stream of {@link Entry} objects
    */
   public static Stream<Entry> getEntriesStream(
       @NotNull NessieApiV1 api,
       @NotNull String ref,
```

Comment:
Just one thing left: `ref` should be of type `Reference`
(Noted it probably at the wrong place: https://github.com/projectnessie/nessie/pull/3431/files#r813583973)

Reply:
yes, see my response here:
https://github.com/projectnessie/nessie/pull/3431#discussion_r813747650
are we ok with it?